analyzers/src/SonarAnalyzer.CSharp/Rules/XXE/XmlReaderSettingsValidator.cs

```diff
@@ -65,29 +68,37 @@ public bool IsUnsafe(InvocationExpressionSyntax invocation, ISymbol settings)
             var objectCreationAssignments = objectCreation.GetInitializerExpressions().OfType<AssignmentExpressionSyntax>();
 
             var propertyAssignments = GetAssignments(invocation.FirstAncestorOrSelf<MethodDeclarationSyntax>())
-                .Where(assignment => IsMemberAccessOnSymbol(assignment.Left, settings, this.semanticModel));
+                .Where(assignment => IsMemberAccessOnSymbol(assignment.Left, settings, semanticModel));
 
             foreach (var assignment in objectCreationAssignments.Union(propertyAssignments))
             {
                 var name = assignment.Left.GetName();
 
-                if (name =="ProhibitDtd" || name == "DtdProcessing")
+                if (name == "ProhibitDtd" || name == "DtdProcessing")
                 {
                     unsafeDtdProcessing = IsXmlResolverDtdProcessingUnsafe(assignment, semanticModel);
+                    if (unsafeDtdProcessing)
+                    {
+                        secondaryLocations.Add(assignment.GetLocation());
+                    }
```

Comment:
I`ll remove the out param but I think we cannot move the if because it would report FP secondary locations. For example:
.XmlResolver = new XmlUrlResolver(); // Secondary
.ProhibitDtd = true; // Secondary FP
.DtdProcessing = DtdProcessing.Parse; // Secondary

Do you agree to leave the if as it was?

Reply:
Yes, you're right